### PR TITLE
docs(README): add note on the requirement of the `strictNullChecks` option

### LINKED ISSUE #708: Document that `strictNullChecks` is required to be enabled
Many issues have been opened due to the fact that the types resolve to `never`. That is because they don't have `strictNullChecks` enabled.

In light of that, I think we should document the fact that this option is required in order to use the types from this repository

### PR DESCRIPTION
**Pull request by wolfy1339**
Fixes #708

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -301,6 +301,8 @@ const handleIssuesOpenedEvent = (event: IssuesOpenedEvent) => {
 };
 ```
 
+**Note**: You _MUST_ enable the [`strictNullChecks` option](https://www.typescriptlang.org/tsconfig#strictNullChecks) in your `tsconfig` in order to use the types. See [#395](https://github.com/octokit/webhooks/issues/395) for details.
```
Comment (oscard0m) [inline README.md:304]: I would try to rephrase this to sound less imperative but looks good to me.

```suggestion
**⚠️ Caution ⚠️ **: Webhooks Types are expected to be used with [`strictNullChecks` option](https://www.typescriptlang.org/tsconfig#strictNullChecks) in your `tsconfig`. If you don't have this option enabled, there's the possibility you get `never` as inferred type in some use cases. See [#395](https://github.com/octokit/webhooks/issues/395) for details.
```

File: payload-types/README.md [hunk 2]
```diff
@@ -30,6 +30,8 @@ const handleIssuesOpenedEvent = (event: IssuesOpenedEvent) => {
 };
 ```
 
+**Note**: You _MUST_ enable the [`strictNullChecks` option](https://www.typescriptlang.org/tsconfig#strictNullChecks) in your `tsconfig` in order to use the types. See [#395](https://github.com/octokit/webhooks/issues/395) for details.
```
Comment (oscard0m) [inline payload-types/README.md:33]: Same as the previous comment.

**Review by oscard0m — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (wolfy1339) [reply README.md]: Updated to use your wording

**Review by oscard0m — APPROVED** [review]

**Review by gr2m — APPROVED** [review]
